Kernel/Syscalls/mmap.cpp

```diff
@@ -417,8 +417,6 @@ ErrorOr<FlatPtr> Process::sys$madvise(Userspace<void*> address, size_t size, int
             if (!region->vmobject().is_anonymous())
                 return EINVAL;
             auto& vmobject = static_cast<Memory::AnonymousVMObject&>(region->vmobject());
-            if (!vmobject.is_purgeable())
```

Comment:
What is the reasoning for this change? It wasn't explained in the commit message?

Reply:
There was a TOCTOU here, because we first checked that the VMObject is purgeable, hence it was needed to grab the lock, and then again call the `set_volatile` method which assume it stays purgeable and it internally takes the lock on the attributes again. Instead, I just ensure the `set_volatile` method checks if it's purgeable VMObject while taking the lock only once, so there's no TOCTOU here.

Reply:
I think by now we all understand that the code is perfectly fine, because it has a check where's that appropriate to actually be located in. I will look into the commit message but overall I think this is really OK.

Reply:
I changed the commit message a bit to ensure the reason for this is written there.